src/player.cpp

```diff
@@ -14442,40 +14444,77 @@ std::vector<mission*> player::get_failed_missions() const
     return failed_missions;
 }
 
-void player::print_encumbrance(WINDOW *win, int min, int max, int line) const
+encumbrance_data player::get_encumbrance(size_t i) const
+{
+    encumbrance_data e;
+    e.iBodyTempInt = (temp_conv[i] / 100.0) * 2 - 100; // Scale of -100 to +100
+    e.iEnc = encumb(bp_aBodyPart[i], e.iLayers, e.iArmorEnc);
+    return e;
+}
+
+void player::print_encumbrance(WINDOW *win, int line) const
 {
+    // which of the four paired body parts should get combined?
+    bool combined_limbs[4] = {}; // all false
+    for (size_t i = 0+4; i<12-1; i+=2) {
+        if (get_encumbrance(i) == get_encumbrance(i+1)) {
+            combined_limbs[i/2-2] = true;
+        }
+    }
+
+    // calculate the top and bottom body parts that fit in the window
+    int min, max;
+    int encumb_win_size_y = 8;
+    int half_y = encumb_win_size_y / 2;    
+    if (line <= half_y) {
+        min = 0;
+    } else if (line >= 12 - half_y) {
+        min = (12 - encumb_win_size_y);
+    } else {
+        min = line - half_y;
+    }
+    max = min + encumb_win_size_y;
+    for(int i=0; i<4; i++) {
+        if (max >= i*2+6 && combined_limbs[i]) {
+            max = max + (min==0);
+            min-=min>0;
+        }
+    }
+
+    draw_scrollbar(win, (line>=0)?line:0, encumb_win_size_y, 12-combined_limbs[3], 1);
+
     // initialize these once, and only once
-    static std::string asText[] = {_("Torso"), _("Head"), _("Eyes"), _("Mouth"), _("L. Arm"), _("R. Arm"),
-                             _("L. Hand"), _("R. Hand"), _("L. Leg"), _("R. Leg"), _("L. Foot"),
-                             _("R. Foot")};
-    static body_part aBodyPart[] = {bp_torso, bp_head, bp_eyes, bp_mouth, bp_arm_l, bp_arm_r, bp_hand_l,
-                             bp_hand_r, bp_leg_l, bp_leg_r, bp_foot_l, bp_foot_r};
-    int iEnc, iArmorEnc, iBodyTempInt;
-    double iLayers;
     std::string out;
     /*** I chose to instead only display X+Y instead of X+Y=Z. More room was needed ***
      *** for displaying triple digit encumbrance, due to new encumbrance system.    ***
      *** If the player wants to see the total without having to do them maths, the  ***
      *** armor layers ui shows everything they want :-) -Davek                      ***/
-    for (int i = min; i < max; ++i) {
+    int bp, row;
+    max = max>12?12:max;
```

Comment:
std::min kept complaining about comparing int to size_t. I forgot to switch back when I got rid of all the size_t.